core/app/components/pubs/PubEditor/actions.ts

```diff
@@ -49,23 +57,38 @@ export const updatePub = defineServerAction(async function updatePub({
 	pubId,
 	pubValues,
 	stageId,
+	formSlug,
 	continueOnValidationError,
 }: {
 	pubId: PubsId;
 	pubValues: PubValues;
 	stageId?: StagesId;
+	formSlug?: string;
 	continueOnValidationError: boolean;
 }) {
 	const loginData = await getLoginData();
 
-	if (!loginData) {
-		throw new Error("Not logged in");
+	if (!loginData || !loginData.user) {
+		return ApiError.NOT_LOGGED_IN;
 	}
 
 	const community = await findCommunityBySlug();
 
 	if (!community) {
-		throw new Error("Community not found");
+		return ApiError.COMMUNITY_NOT_FOUND;
+	}
+
+	const canUpdateFromForm = formSlug
+		? await userHasPermissionToForm({ formSlug, userId: loginData.user.id })
```

Comment:
I guess this isn't changing the situation but this allows anyone who has been invited to a form to edit any pub in the community with that form. I know you didn't introduce that problem but could you add a TODO comment here?

Reply:
okay added a TODO for now!